client/src/modules/authorization/components/SignInForm.jsx

```diff
@@ -0,0 +1,206 @@
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const newState = { pending: false };
+
+          if (err instanceof UnauthorizedException) {
+            newState.signInErrorId =
+              'authorization.actions.sign-in.invalid-credentials';
+          } else {
+            newState.signInErrorId = 'common.something-went-wrong';
+          }
+
+          this.setState(newState);
+        }
+      });
+  };
+
+  renderSignUpError = () => {
+    const { signInErrorId } = this.state;
+    const {
+      intl: { formatMessage }
+    } = this.props;
+
+    return (
+      <p className="sign-in__error">
+        <FormattedMessage
+          id={signInErrorId}
+          values={{
+            link: (
+              <Link className="sign-in__link" to="/reset-password">
+                {formatMessage({ id: 'authorization.forgot-password' })}
+              </Link>
+            )
+          }}
+        />
+      </p>
+    );
+  };
+
+  render() {
+    const { isFormValid, pending, signInErrorId } = this.state;
+    const {
+      intl: { formatMessage },
+      onCancel
+    } = this.props;
+    const hasSignUpFailed = signInErrorId.length > 0;
+
+    return (
+      <div className="sign-in">
+        <h1 className="sign-in__heading">
+          <FormattedMessage id="authorization.auth-box.sign-in" />
+        </h1>
+        {signInErrorId && this.renderSignUpError()}
+        <form className="sign-in__form" noValidate>
+          <AuthInput
+            disabled={pending}
+            formError={hasSignUpFailed}
+            focus
+            labelId="authorization.input.email.label"
+            name="email"
+            onChange={this.onEmailChange}
+            type="text"
+            validator={this.emailValidator}
+          />
+          <AuthInput
+            disabled={pending}
+            formError={hasSignUpFailed}
+            labelId="authorization.input.password.label"
+            name="password"
+            onChange={this.onPasswordChange}
+            type="password"
+          />
+          <div className="sign-in__buttons">
+            <button
+              className="primary-button"
+              disabled={pending}
+              onClick={onCancel}
+              type="button"
+            >
+              <FormattedMessage id="common.button.cancel" />
+            </button>
+            <PendingButton
+              className="primary-button sign-in__confirm"
+              disabled={!isFormValid}
+              onClick={this.handleSignIn}
+              type="button"
+            >
+              <FormattedMessage id="authorization.sign-in" />
+            </PendingButton>
+          </div>
+        </form>
+        <p className="sign-in__forgot-password">
+          <Link className="sign-in__link" to="/reset-password">
+            {formatMessage({ id: 'authorization.forgot-password' })}
+          </Link>
+          ?
```

Comment:
do not create questions like this. It might work in this language but fail in others. Create separate strings for the question and forgot password link.